### DISCUSSION TITLE
Cache always missed

### DISCUSSION THREAD

**Discussion by romanoff** [vercel/turborepo#1214] [Help] [answered]
Likely am missing something. Set up workspaces with pnpm and turborepo. Here are a few problems that I am facing:
- pnpm build always returns cache miss for app. I checked sha1sum of generated index.js and it is same between builds
Example:
```
 Packages in scope: adder, app
• Running build in 2 packages
app:build: cache miss, executing 37a302a8fbf4f25a
app:build: 
app:build: > app@1.0.0 build /home/aromanov/src/experiments/js/workspaces-example/apps/app
app:build: > esbuild src/index.ts --bundle  --platform=node --outfile=dist/index.js
app:build: 
app:build: 
app:build:   dist/index.js  1.2kb
app:build: 
app:build: ⚡ Done in 2ms

 Tasks:    1 successful, 1 total
Cached:    0 cached, 1 total
  Time:    481ms 

```
- Would it be possible to use `turborepo build` for root index.ts that is not part of the workspaces? (it has dependencies to some of workspace packages)

Attached simple code that I was using to understand turborepo better

[workspaces-example.zip](https://github.com/vercel/turborepo/files/8660128/workspaces-example.zip)

**Comment by ToyB0x** [comment, marked as answer]
Looking at your .gitignore file doesn't seem to ignore the generated folders such as `.turbo`.
(This folder contains turbo-build.log, which changes with each build.)

Reference: [Official documentation](https://turborepo.org/docs/core-concepts/caching#hashing)
> Hash the contents of all not-gitignored files in the package folder or the files matching the inputs globs, if present

I hope this helps you 📝

**Reply by ToyB0x** [reply]
@romanoff 

FYI: Using the following .gitignore file will cache successfully 🚀 
```
.turbo
```

![image](https://user-images.githubusercontent.com/83337042/171769034-51d6cf3a-7953-4728-a1d5-91de08c36ba7.png)

**Comment by arantespp** [comment]
I had this problem because my `build` command always changed files without lint rules or auto-generated docs. Running `git checkout -- .` solved for me. https://github.com/ttoss/ttoss/pull/100
